### ISSUE TITLE
Gnome doesn't recognize newly created window

### ISSUE THREAD

**Issue by Naadiyaar** [alacritty/alacritty#6931]
There's a bug I'm facing on gnome when launching Alacritty at overview mode (can see workspaces and dock) and open a terminal window gnome doesn't recognize it and I can't send any command to the session, unless clicking on the window by mouse or open a new window and close it, so focus goes on the previous one automatically. This happens both when I open it using key combo or clicking on the icon in dock.

Fedora workstation 38
alacritty 0.12.0
wayland, gnome 44.1

this is a general log, not for time I'm facing the issue (don't know where to get that)
```
alacritty -vv
Created log file at "/tmp/Alacritty-4466.log"
[0.000006705s] [INFO ] [alacritty] Welcome to Alacritty
[0.000246329s] [INFO ] [alacritty] Version 0.12.0
[0.000260018s] [INFO ] [alacritty] Running on Wayland
[0.000830618s] [INFO ] [alacritty] Configuration files loaded from:
                                     "/home/naadi/.config/alacritty/alacritty.yml"
[0.048616204s] [INFO ] [alacritty] Using EGL 1.5
[0.073993187s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Bold, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[0.085880672s] [INFO ] [alacritty] Window scale factor: 1
[0.087766587s] [DEBUG] [alacritty] Loading "Iosevka Nerd Font" font
[0.090229178s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Regular, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[0.093522109s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Bold, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[0.096928530s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Italic, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[0.100127524s] [DEBUG] [crossfont] Loaded Face Face { ft_face: Font Face: Bold Italic, load_flags: TARGET_LIGHT, render_mode: "Normal", lcd_filter: 1 }
[0.113840512s] [INFO ] [alacritty] Running on AMD Radeon Graphics (renoir, LLVM 16.0.1, DRM 3.49, 6.2.14-300.fc38.x86_64)
[0.113891147s] [INFO ] [alacritty] Using OpenGL 3.3 renderer
[0.122720159s] [DEBUG] [alacritty] Filling glyph cache with common glyphs
[0.163958925s] [INFO ] [alacritty] Cell size: 8 x 21
[0.164023319s] [INFO ] [alacritty] Padding: 5 x 5
[0.164031979s] [INFO ] [alacritty] Width: 810, Height: 598
[0.164097769s] [INFO ] [alacritty] PTY dimensions: 28 x 100
[0.168532598s] [INFO ] [alacritty] Initialisation complete
[0.215367649s] [INFO ] [alacritty] Font size changed to Size(24) with scale factor of 1
[0.215435884s] [INFO ] [alacritty] Cell size: 8 x 21
[0.215470036s] [DEBUG] [alacritty_terminal] Term::resize dimensions unchanged
[0.221226815s] [INFO ] [alacritty] Padding: 5 x 5
[0.221282268s] [INFO ] [alacritty] Width: 810, Height: 598
```

**Comment by chrisduerr** [comment]
Considering that this sounds extremely specific to Gnome's window management behavior, what makes you think this is an Alacritty bug?

**Comment by Naadiyaar** [comment]
Since I'm able to run other applications with the same situation but send them input without any problem, we thought this might be a problem in alacritty.

**Comment by pvizc** [comment]
I've noticed this behavior with some other apps as well, however, Gnome native apps always get focus, so I think it is correct to assume that this has something to do with the implementation or the communication with the DE.

**Comment by kchibisov** [comment]
GNOME changed the behavior and now requires xdg activation startup. I haven't figured out what exactly it wants from it though.

**Comment by GregTheHun** [comment]
I'm getting the same issue, it is strange, but I wonder if there's a workaround that would try to grab focus once opening the app, depending on the DE?

**Comment by kchibisov** [comment]
It'll be solved with the upcoming release, but I don't think there's a workaround for that at the moment. There's also no ETA when the fix for that will even land (requires upstream work).

Basically GNOME wants us to use special startup protocol, which we support, but we don't support starting with it. I know that on my fedora 38 workstation it wasn't happening yet, though, maybe the issue is with how I've installed alacritty on it...

**Comment by kchibisov** [comment]
Could someone try editing their `Alacritty.desktop` file and remove the `StartupWMClass=Alacritty` from it?

Alternatively you could try https://github.com/alacritty/alacritty/pull/7072 but you **must** update the `.desktop` file with the one is used on the branch I've linked.
Otherwise it all may not work.

**Comment by Naadiyaar** [comment]
> Could someone try editing their Alacritty.desktop file and remove the StartupWMClass=Alacritty from it?

so i created a user-wide desktop entry `.local/share/applications/Alacritty.desktop` and removed `StartupWMClass=Alacritty` from it, i did a reboot after that but problem still exist
alacritty 0.12.2

**Comment by kchibisov** [comment]
Well, then you have an option to try the PR I've posted. But ensure that gnome will read the desktop entry from it.

**Comment by Naadiyaar** [comment]
> Well, then you have an option to try the PR I've posted. But ensure that gnome will read the desktop entry from it.

you mean i should pull the branch and build it myself to test? is that necessary? because i have no experience with Cargo and Rust in general

**Comment by kchibisov** [comment]
Well, that's the only option to know whether it's enough and if you want to get this issue fixed, because on my fedora 38 gnome 44.2 this issue is not happening even without this patch.

From what I've understood it should help, but maybe someone else on this issue could test this as well.

**Comment by chrisduerr** [comment]
> you mean i should pull the branch and build it myself to test? is that necessary? because i have no experience with Cargo and Rust in general

No experience should be required. Our installation instructions should allow anyone to build from source.

**Comment by Naadiyaar** [comment]
By following install.md instruction now I have an Alacritty binary but i doubt if i've done one part right, so i checked out to `next-winit-update` branch and ran `cargo build --release` in this case will Cargo compile wanted branch? if yes, then we're ok, if not tell me how should i do it.

however, the issue is still present even after installing .desktop file from #7072 when i open Alacritty in GNOME's overview mode, focuses doesn't happen automatically
i'm doing this on aa Fedora 38 VM with gnome 44.1

**Comment by Naadiyaar** [comment]
I must add probably this issue isn't related to Alacritty and opening it here was wrong in the first place, it seems this problem exist for every terminal emulator on Wayland, as i checked gnome-terminal, gnome-console and kitty also show same behavior on Wayland.

**Comment by kchibisov** [comment]
Hm, if `gnome-terminal` doesn't work, it likely means something inside the gnome and we can't do anything about it. I'd suggest to handle that upstream.

I thought that the initial issue was due to activation, and we've added activation as well as we've updated the `.desktop` files to say that we support window activation. However if gnome-terminal doesn't work, when it uses gtk and which supports activation automatically, then it's something else, probably a race within gnome's compositor.
